tests/ecosystem/upgrade/conftest.py

```diff
@@ -0,0 +1,309 @@
+import logging
+
+from ocs_ci.ocs import constants, resources
+import pytest
+import threading
+
+log = logging.getLogger(__name__)
+
+
+def create_pods(
+    interface,
+    pvc_factory,
+    pod_factory,
+    storageclass,
+    count,
+    access_mode,
+    volume_mode=None
+):
+    """
+    Create pods for upgrade testing. pvc_factory and pod_factory have to be
+    in the same scope.
+
+    Args:
+        interface (str): CephBlockPool or CephFileSystem
+        pvc_factory (function): Function for creating PVCs
+        pod_factory (function): Function for creating pods
+        storageclass (obj): Storageclass to use
+        count (int): Number of pods to create
+        access_mode (str): ReadWriteOnce, ReadOnlyMany or ReadWriteMany.
+            This decides the access mode to be used for the PVC
+        volume_mode (str): Volume mode for rbd RWO PVC
+
+    Return:
+        list: List of generated pods
+    """
+    log.info(
+        f"Creating {count} pods via {interface} using {access_mode}"
+        f" access mode, {volume_mode} volume mode and {storageclass.name}"
+        f" storageclass"
+    )
+    pvcs = [
+        pvc_factory(
+            storageclass=storageclass,
+            access_mode=access_mode,
+            volume_mode=volume_mode
+        ) for _ in range(count)
+    ]
+    if volume_mode == constants.VOLUME_MODE_BLOCK:
+        pod_dict = constants.CSI_RBD_RAW_BLOCK_POD_YAML
+        raw_block_pv = True
+    else:
+        pod_dict = ''
+        raw_block_pv = False
+    pods = [
+        pod_factory(
+            interface=interface,
+            pvc=pvc,
+            raw_block_pv=raw_block_pv,
+            pod_dict_path=pod_dict,
+        ) for pvc in pvcs
+    ]
+    return pods
+
+
+@pytest.fixture(scope='session')
+def default_storageclasses(request, teardown_factory_session):
```

Comment:
Is there reason have such general fixture available just for upgrade? Maybe worth to move it to /tests/conftest.py WDYT?

Reply:
Sure